### ISSUE TITLE
rename?

### ISSUE THREAD

**Issue by frolic** [holic/web3-scaffold#28]
It was `web3-dapp-scaffold`, then `web3-app-scaffold` (because I didn't feel right calling it a decentralized app), and now `web3-scaffold`.

I'm not a massive fan of the "scaffold" word, feels a little obscure, borrowed from Ruby(?), and probably hurts word of mouth long term?

Maybe `create-web3-app`?

**Comment by davisshaver** [comment]
I see your points about `scaffold`, although I happen to like it as a name. I also like the idea of `create-web3-app` although my expectation for `create-*` projects is that there's a CLI interface to get started. Maybe this is the direction you want to head? In terms of other synonyms for scaffold, I think "starter" or "starter kit" is a common one. Curious to hear what others think about this too.

**Comment by jamiew** [comment]
I had the same thought re. `create-*` pattern -- arguably we could add the CLI bits? Is that worth the work?

I have no qualms with `web3-scaffold` too, which is self-descriptive. I also miss ruby though

**Comment by frolic** [comment]
Yeah, my thought with `create-web3-app` was that it'd be neat to have support for `npm init web3-app` (and potentially other CLI options). Could potentially make it pluggable, e.g. make Tailwind optional.

Looks like someone snagged this name a few years back though: https://www.npmjs.com/package/create-web3-app

**Comment by frolic** [comment]
Also just came across https://www.npmjs.com/package/create-web3, which lets you pick Next/Vite, Hardhat/Foundry, etc. Pretty neat!

**Comment by davisshaver** [comment]
+1 to all of this. A CLI interface would be very cool at some point.

I wonder if there's an architecture that would let us keep this repo (maybe converting to a [template repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) at some point?) and build a CLI helper at a future date which wraps the repo and makes it easy to customize things.

**Comment by frolic** [comment]
> maybe converting to a [template repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) at some point?

Fun fact: this was a template repo! but github nukes commit history and it was hard to port changes over, so I moved it back to a regular repo that you'd fork

**Comment by e-roy** [comment]
first thing I think of when I hear scaffold is [scaffold-eth](https://github.com/scaffold-eth/scaffold-eth).  not suggesting a change in name, just saying it does resonate in the web3 space

**Comment by jamiew** [comment]
eyyy a wild create-web3 author appears :) welcome @e-roy

**Comment by frolic** [comment]
Chatted with the previous owner of create-web3-app and he generously handed over the name! https://www.npmjs.com/package/create-web3-app

**Comment by jamiew** [comment]
Wow, huge!

**Comment by cjpais** [comment]
> +1 to all of this. A CLI interface would be very cool at some point.
> 
> I wonder if there's an architecture that would let us keep this repo (maybe converting to a [template repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) at some point?) and build a CLI helper at a future date which wraps the repo and makes it easy to customize things.

I might have created an issue for this accidentally, hadn't read this thread yet

https://github.com/holic/web3-scaffold/issues/40

**Comment by samkingco** [comment]
Yeah I’d also love a CLI, even if it just copies the template, inits a new git repo, and renames all the packages to my project name. I thought about putting one together next week. Should be quite straightforward.

Right now I have to fork/clone the repo and then nuke history and do a bunch of find and replace. Having one command would be great, and the create-web3-app fits that mental model better.
